src/qibocal/protocols/utils.py

```diff
@@ -757,39 +762,253 @@ def table_html(data: dict) -> str:
+    #   W_zca = E * D * E.T
+    W = np.dot(np.dot(E, D), E.T)
+
+    X_white = np.dot(X, W)
+
+    return X_white
+
+
+def filter_data(matrix_z: np.ndarray):
+    """Filter data with a ZCA transformation and then a unit-variance Gaussian."""
+
+    # adding zca filter for filtering out background noise gradient
+    zca_z = zca_whiten(matrix_z)
+    # adding gaussian fliter with unitary variance for blurring the signal and reducing noise
+    return ndimage.gaussian_filter(zca_z, 1)
+
+
+def scaling_global(sig: np.ndarray) -> np.ndarray:
+    """Min–max scaling over the whole np.ndarray (global)."""
+    return scaling_slice(sig, axis=None)
+
+
+def scaling_slice(sig: np.ndarray, axis: Optional[int]) -> np.ndarray:
+    """Min–max scaling over a specific axis of the np.ndarray."""
+
+    def expand(a):
+        return np.expand_dims(a, axis) if axis is not None else a
+
+    sig_min = expand(np.min(sig, axis=axis))
+    return (sig - sig_min) / (expand(np.max(sig, axis=axis)) - sig_min)
+
+
+def horizontal_diagonal(xs: np.ndarray, ys: np.ndarray) -> float:
+    """Computing the lenght of the diagonal of a two dimensional grid."""
+    sizes = np.empty(2)
+    for i, values in enumerate([xs, ys]):
+        sizes[i] = np.max(values) - np.min(values)
+    return np.sqrt((sizes**2).sum())
+
+
+def build_clustering_data(peaks_dict: dict, z: np.ndarray):
+    """Preprocessing of the data to cluster."""
+    x_ = peaks_dict["x"]["idx"]
+    y_ = peaks_dict["y"]["idx"]
+    z_ = z[y_, x_]
+
+    rescaling_fact = horizontal_diagonal(x_, y_) / 10
+    global DISTANCE
+    DISTANCE = 1.5 * rescaling_fact  # very heuristic
+    """Minimum distance for separate clusters.
+
+    Clusters below this distance will be merged.
+    Since it is given in a 3D-space, with a compressed vertical dimension, and the horizontal plane measured in pixels, this distance correspond to diagonally adjacent pixels, with some additional leeway for the extra dimension.
+    """
+    return np.stack((x_, y_, scaling_global(z_) * rescaling_fact)).T
+
+
+def peaks_finder(x, y, z) -> dict:
+    """Function for finding the peaks over the whole signal.
 
-    Given a dataset of the form (x, y, z) where a spike or a valley is expected,
-    this function discriminate the points (x, y) with a signal, from the pure noise
-    and return the first ones.
+    This function takes as input 3 features of the signal.
+    It slices the dataset along a preferred direction (`y` dimension, corresponding to the flux bias) and for each slice it determines the biggest peaks
+    by using `scipy.signal.find_peaks` routine.
+    It returns a dictionary `peaks_dict` containing all the features for the computed peaks.
     """
+
+    # filter data using find_peaks
+    peaks = {"x": {"idx": [], "val": []}, "y": {"idx": [], "val": []}}
+    for y_idx, y_val in enumerate(y):
+        signal_fixed_y = z[y_idx]
+        peak, info = find_peaks(signal_fixed_y, prominence=0.2)
```

Comment:
At this point, the name is more confusing than helpful
```suggestion
        peak, info = find_peaks(z[y_idx], prominence=0.2)
```

If it was used more than once, the assignment was almost mandatory. And even when it is used once, it may be worth.
But in this case, the name is not really clarifying that much the value. And it is possibly more confusing.